src/DetailView/DetailView.tsx

```diff
@@ -1,19 +1,139 @@
 import React, { FunctionComponent } from 'react';
-import { FullPokemonAttributes } from '../types/index';
+import { FullPokemonAttributes, Type, Stat, Move } from '../types/index';
+import './DetailView.scss';
+import pokeload from '../assets/pikapokeball.gif';
 
 interface DetailViewProps {
   pokemon: FullPokemonAttributes;
 }
 
-const DetailView: FunctionComponent<DetailViewProps> = ({ pokemon }) => (
-  <div className="view">
-    <h1>
-      #{pokemon.pokemon_id} {pokemon.name}
-    </h1>
-    <img alt={pokemon.name} src={pokemon.sprites.front_default} />
-    <h3>Height: {pokemon.height} decimeters</h3>
-    <h3>Weight: {pokemon.weight} hectograms</h3>
-  </div>
-);
+const typeStyleName = (types: Array<Type>) => {
+  const prefix = 'type-' + (types.length === 1 ? 'single' : 'double') + '-';
+  return prefix + types.map(type => type.type.name).join('-');
+};
+
+const typeHeader = (types: Array<Type>) => {
+  return types.map(type => (
+    <div className={`type-header-${type.type.name} view-header`}>
+      {type.type.name}
+    </div>
+  ));
+};
+
+const heightConverter = (height: number) => {
```

Comment:
Some of these are good candidates to move into a utils or helpers file. Make a `src/utils.ts` or `src/helpers.ts` (dealer's choice).

Reply:
Good point! Done.